### PR TITLE
Feature/pm 422 add motor voltage

### PR DESCRIPTION
**Pull request by Ciliaclaij**
Closes PM-422

## Description
Add the reading of the motor voltage for linear joint control

## Changes
e.g.
* Added motor voltage reading
* Added message to publish motor voltage
* Merged PM-436 into this branch to use subindex PDO reading

### CODE REVIEW

File: march_hardware/src/IMotionCube.cpp
Comment (RutgerVanBeek) [inline march_hardware/src/IMotionCube.cpp]: DCLinkVoltage is the other voltage or am I wrong? At least this is the same object as used in the `getIMCVoltage()`

**Review by RutgerVanBeek — CHANGES_REQUESTED** [review]
I think you know this better what the `DCLinkVoltage` represents. Currently the `getIMCVoltage()` and `getMotorVoltage()` point both to the `DCLinkVoltage`. You should add an object to the PDO and use that object for one of the two.

File: march_hardware/src/IMotionCube.cpp
Comment (Ciliaclaij) [reply march_hardware/src/IMotionCube.cpp]: That was a mistake with copying the lines, after fixing all the merge conflicts :(

**Comment by RutgerVanBeek** [comment]
There was an error when writing the sdo's on startup, see Slack for more information. (https://projectmarch.slack.com/archives/CJYPA42UC/p1589362128001700)

**Comment by Ishadijcks** [comment]
Maybe provide a link to the relevant Slack conversation?

**Review by RutgerVanBeek — DISMISSED** [review]

**Review by Joris-JW — COMMENTED** [review]
I would wait with the merge of this branch until PM 436 is merged to develop, chronological speaking that would be most logical.

**Comment by RutgerVanBeek** [comment]
Oh I remember something now, @Ciliaclaij where you planning on converting the values to more understandable units or is that for another issue?

**Comment by Ciliaclaij** [comment]
@RutgerVanBeek I realised this, when I work again on monday I will implement this. So it's better to wait for that (and PM 436) to merge this

File: march_hardware/src/IMotionCube.cpp
Comment (Olavhaasie) [inline march_hardware/src/IMotionCube.cpp:307]: Are you aware that the `.ui` at the end means that the value is read as an unsigned 16 bit integer? I ask this because the return type of the function is `float`. To get the `float` value you would have to do `.f`.

Comment (RutgerVanBeek) [reply march_hardware/src/IMotionCube.cpp:307]: This is probably the conversion issue I mention above.

Comment (RutgerVanBeek) [reply march_hardware/src/IMotionCube.cpp:307]: You read only 16bit so you cannot make a float out of it. It is a bit strange that they say you have to read 16 bits, but that it is a float. That seems a mistake on their side.

File: march_hardware/src/IMotionCube.cpp
Comment (Olavhaasie) [inline march_hardware/src/IMotionCube.cpp]: As Rutger already mentioned, this is not possible since float can only be a 32 bit value. You would have to do some manual conversions. You should convert this back for now and we'll see later about the conversion.
```suggestion
  return this->read16(this->miso_byte_offsets_.at(IMCObjectName::MotorVoltage)).ui;
```

**Review by Olavhaasie — APPROVED** [review]

**Review by RutgerVanBeek — APPROVED** [review]

**Pull request merged**
